packages/graphql/src/schema/parse-node-directive.ts

```diff
@@ -37,4 +67,8 @@ function getArgumentValue<T>(directive: DirectiveNode, name: string): T | undefi
     return argument ? (valueFromASTUntyped(argument.value) as T) : undefined;
 }
 
+function alphabeticallySortFields(fields: FieldDefinitionNode[]): FieldDefinitionNode[] {
+    return fields.sort((a, b) => (a.name.value > b.name.value ? 1 : -1));
+}
```

Comment:
**question:** just a quick thought on how most people will grow their schema... It would be cool if between server restarts, and adding fields to types, the global IDs try their best to remain consistent. By sorting alphabetically, if a user adds a new unique field beginning the letter "a", it will automatically become the global ID. As users are quite likely to add new fields to the bottom of a type, are we better taking the first unique or ID field in the type as it is written? I would say that's more likely to remain at the top for longer, and remain a more consistent global ID. Thoughts?

Reply:
The cleanest way to do this would be to have the user define the field in the directive. Then we could do a check to make sure that the field is decorated with the @unique or @id directives. 

There are directives where field names are used. This seems to be a good use case -- no need to infer something so important when the user can just define it.

Reply:
Potentially... How do you propose we do this?

```gql
type Book @node(globalId: "isbn") {
  isbn: String! @unique
}
```

I think we discussed this before, and we didn't like it because it wasn't 100% apparent that this was a toggle for behaviour.

```gql
type Book @node(global: { enabled: true, id: "isbn" }) {
  isbn: String! @unique
}
```

Perhaps? Appreciate your thoughts!

Reply:
Hmm, I'm probably in favor of the first -- I don't see a world in which someone wants to toggle global nodes but not set the idField. In fact, it will err if they don't, so I just see it as adding an unnecessary variable. 

But I get your hesitation, so I have no objection to 2, especially if the alternative is to try and infer it based on the order of the schema. But let me know what you think. This would be a breaking change if implemented later on, so we should probably come to a decision before adding the feature.